### PR TITLE
Qe branch 3

### PR DESCRIPTION
**Pull request by deads2k**
I think I pushed.

### CODE REVIEW

File: pkg/api/component_report.go [hunk 1]
```diff
@@ -742,6 +742,10 @@ func (c *componentReportGenerator) normalizeProwJobName(prowName string) string
 	if prev, err := previousRelease(c.SampleRelease.Release); err == nil {
 		name = strings.ReplaceAll(name, prev, "X.X")
 	}
+	name = strings.ReplaceAll(name, "f28", "fXX")
+	name = strings.ReplaceAll(name, "f14", "fXX")
+	name = strings.ReplaceAll(name, "f7", "fXX")
+	name = strings.ReplaceAll(name, "f6", "fXX")
 	return name
```
Comment (stbenjam) [inline pkg/api/component_report.go:749]: Will they have other frequencies?  I know you hate regex but this isn't too bad:

	re := regexp.MustCompile(`-f\d+`)
	output := re.ReplaceAllString(input, "-fXX")

File: cmd/sippy/component_readiness.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+	}
+
+	cacheClient, err := f.CacheFlags.GetCacheClient()
+	if err != nil {
+		return errors.WithMessage(err, "couldn't get cache client")
+	}
+
+	var bigQueryClient *bigquery.Client
+	var gcsClient *storage.Client
+	if f.GoogleCloudFlags.ServiceAccountCredentialFile != "" {
```
Comment (deads2k) [inline cmd/sippy/component_readiness.go:127]: I'd like to push this logic into the GoogleCloudFlags type, so we can be consistent with how to get output from the input.

File: pkg/sippyserver/install_details.go [hunk 3]
```diff
@@ -7,12 +7,22 @@ import (
 )
 
 func (s *Server) jsonUpgradeReportFromDB(w http.ResponseWriter, req *http.Request) {
```
Comment (deads2k) [inline pkg/sippyserver/install_details.go:9]: Overall, would we try to refactor Server to have a componentREadinessServer?

File: pkg/sippyserver/install_details.go [hunk 4]
```diff
@@ -7,12 +7,22 @@ import (
...
+		return
+	}
+
 	release := req.URL.Query().Get("release")
 
 	api.PrintUpgradeJSONReportFromDB(w, req, s.db, release)
 }
 
 func (s *Server) jsonInstallReportFromDB(w http.ResponseWriter, req *http.Request) {
+	if s.db == nil {
```
[the remark this answers is not in the document]
Comment (stbenjam) [reply pkg/sippyserver/install_details.go:21]: There's an if block in the server section that checks for s.db.  Leftover from when we were partially on testgrid disk data and partially on postgres.  You could move all these under there.

**Comment by stbenjam** [comment]
I don't think this has to be perfect to merge, it's making progress and would keep you from having to maintain a separate branch.  I think we should just merge this once lint passes.

File: pkg/sippyserver/install_details.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (stbenjam) [reply pkg/sippyserver/install_details.go:21]: I refactored this quite a bit in https://github.com/openshift/sippy/pull/1559, it includes most of this PR

**Pull request closed without merging**